lib/vrf.c

```diff
@@ -997,17 +997,29 @@ const char *vrf_get_default_name(void)
 int vrf_bind(vrf_id_t vrf_id, int fd, const char *name)
 {
 	int ret = 0;
+	struct vrf *vrf;
 	struct interface *ifp;
 
-	if (fd < 0 || name == NULL)
+	if (fd < 0 || name == NULL || vrf_is_backend_netns()
+	    || (strcmp(name, "default") == 0) || vrf_id == VRF_UNKNOWN)
 		return fd;
-	/* the device should exist
```

Comment:
- I think this code can be simplified.
just check for vrf_identifier value.
if vrf_id == VRF_UNKNOWN, stop processing, return -1.
- also not sure some extra code related to netns should be handled for now